### PR TITLE
refactor: use split instead of `serde_urlencoded` in http auth

### PR DESCRIPTION
**Pull request by shuiyisong**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Currently authorization in http api uses `serde_urlencoded` to decode query string. Since we only need `db` parameter, 
it's more performant and simple to just use `split` and extract `db` key.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

File: src/servers/src/http/authorize.rs [hunk 1]
```diff
@@ -157,18 +194,8 @@ fn get_influxdb_credentials<B: Send + Sync + 'static>(
...
-            Err(e) => IllegalParamSnafu {
-                msg: format!("fail to parse http query: {e}"),
-            }
-            .fail()?,
-        };
-
-        let username = query.get("u");
-        let password = query.get("p");
+        let username = extract_helper(query_str, &regex_extractor.influxdb_user_extractor);
+        let password = extract_helper(query_str, &regex_extractor.influxdb_pass_extractor);
```
Comment (sunng87) [inline src/servers/src/http/authorize.rs:198]: if we want extreme performance we can combine these user/pass regex into one and match only once

File: src/servers/src/http/authorize.rs [hunk 2]
```diff
@@ -367,4 +395,70 @@ mod tests {
...
+        let re = Regex::new(INFLUXDB_USER_PATTERN);
+        assert!(re.is_ok());
+        let re = Regex::new(INFLUXDB_PASS_PATTERN);
+        assert!(re.is_ok());
+    }
+
+    #[test]
+    fn test_regex_capture() {
+        let extractor = RegexExtractor::new();
+        matches!(extract_helper("name=bar", &extractor.db_extractor), None);
```
Comment (sunng87) [inline src/servers/src/http/authorize.rs:412]: can we add a test for `db=&name=bar`

**Review by MichaelScofield — APPROVED** [review]

**Comment by MichaelScofield** [comment]
Approved, though I don't think it's best to use regex here. Convert url string to a parameter map is a common and more extensible way to handle parameters. Do we really need to squeeze that performance?

File: src/servers/src/http/authorize.rs [hunk 3]
```diff
@@ -281,8 +258,33 @@ fn need_auth<B>(req: &Request<B>) -> bool {
     path.starts_with(HTTP_API_PREFIX)
 }
 
+fn extract_db_from_query(query: &str) -> Option<&str> {
+    for pair in query.split('&') {
+        if let Some(db) = pair.strip_prefix("db=") {
+            return if pair.len() == 3 { None } else { Some(db) };
```
Comment (sunng87) [inline src/servers/src/http/authorize.rs:264]:
```suggestion
            return if db.is_empty() { None } else { Some(db) };
```

File: src/servers/src/http/authorize.rs [hunk 4]
```diff
@@ -281,8 +258,33 @@ fn need_auth<B>(req: &Request<B>) -> bool {
...
+    None
+}
+
+fn extract_influxdb_user_from_query(query: &str) -> (Option<&str>, Option<&str>) {
+    let mut username = None;
+    let mut password = None;
+
+    for pair in query.split('&') {
+        if pair.starts_with("u=") && pair.len() > 2 {
+            username = Some(&pair[2..]);
```
Comment (sunng87) [inline src/servers/src/http/authorize.rs:276]: Can we use `strip_prefix` like how we parse `db`?

Comment (shuiyisong) [reply src/servers/src/http/authorize.rs:276]:
```Rust
        if let Some(u) = pair.strip_prefix("u=") {
            username = if u.is_empty() { None } else { Some(u) };
        } else if let Some(p) = pair.strip_prefix("p=") {
            password = if p.is_empty() { None } else { Some(p) };
        }

**Review by sunng87 — APPROVED** [review]

**Pull request merged**